### ISSUE TITLE
Bug: Events are firing twice when we have dynamic and static mode

### ISSUE THREAD

**Issue by danielbarion** [pixijs/pixijs#10441]
### Current Behavior

When we have `eventMode = 'static'` and `eventMode = 'dynamic'` in the same app, the events fire twice.

### Expected Behavior

When we have `eventMode = 'static'` and `eventMode = 'dynamic'` in the same app, the events should fire only once.

### Steps to Reproduce

1. Go to https://pixijs.com/8.x/playground?exampleId=events.logger
2. On the following code:
```
    // Enable interactivity everywhere!
    app.stage.eventMode = 'static';
    app.stage.hitArea = app.screen;
    whiteBox.eventMode = 'static';
    blackBox.eventMode = 'static';
```
3. Change any of the event modes to be dynamic:

```
    // Enable interactivity everywhere!
    app.stage.eventMode = 'static';
    app.stage.hitArea = app.screen;
    whiteBox.eventMode = 'dynamic';
    blackBox.eventMode = 'static';
```

Wait it gets reflected into the playground, the events are now firing twice.

[The playground already edited](https://pixijs.com/8.x/playground?state=eyJmaWxlcyI6eyIvc3JjL2luZGV4LmpzIjp7ImNvZGUiOiJpbXBvcnQgeyBBcHBsaWNhdGlvbiwgQXNzZXRzLCBUZXh0LCBHcmFwaGljcyB9IGZyb20gJ3BpeGkuanMnO1xuXG4oYXN5bmMgKCkgPT5cbntcbiAgICAvLyBDcmVhdGUgYSBuZXcgYXBwbGljYXRpb25cbiAgICBjb25zdCBhcHAgPSBuZXcgQXBwbGljYXRpb24oKTtcblxuICAgIC8vIEluaXRpYWxpemUgdGhlIGFwcGxpY2F0aW9uXG4gICAgYXdhaXQgYXBwLmluaXQoeyBhbnRpYWxpYXM6IHRydWUsIGJhY2tncm91bmQ6ICcjMTA5OWJiJywgcmVzaXplVG86IHdpbmRvdyB9KTtcblxuICAgIC8vIEFwcGVuZCB0aGUgYXBwbGljYXRpb24gY2FudmFzIHRvIHRoZSBkb2N1bWVudCBib2R5XG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmRDaGlsZChhcHAuY2FudmFzKTtcblxuICAgIGNvbnN0IHRpdGxlID0gYXBwLnN0YWdlLmFkZENoaWxkKFxuICAgICAgICBuZXcgVGV4dCh7XG4gICAgICAgICAgICB0ZXh0OiBgTW92ZSB5b3VyIG1vdXNlIHNsb3dseSBvdmVyIHRoZSBib3hlcyB0b1xuICAgICAgICBzZWUgdGhlIG9yZGVyIG9mIHBvaW50ZXJlbnRlciwgcG9pbnRlcmxlYXZlLFxuICAgICAgICBwb2ludGVyb3ZlciwgcG9pbnRlcm91dCBldmVudHMgb24gZWFjaCB0YXJnZXQhYCxcbiAgICAgICAgICAgIHN0eWxlOiB7XG4gICAgICAgICAgICAgICAgZm9udFNpemU6IDE2LFxuICAgICAgICAgICAgfSxcbiAgICAgICAgfSksXG4gICAgKTtcblxuICAgIHRpdGxlLnggPSAyO1xuXG4gICAgY29uc3QgbG9ncyA9IFtdO1xuICAgIGNvbnN0IGxvZ1RleHQgPSBhcHAuc3RhZ2UuYWRkQ2hpbGQoXG4gICAgICAgIG5ldyBUZXh0KHtcbiAgICAgICAgICAgIHRleHQ6ICcnLFxuICAgICAgICAgICAgc3R5bGU6IHtcbiAgICAgICAgICAgICAgICBmb250U2l6ZTogMTQsXG4gICAgICAgICAgICB9LFxuICAgICAgICB9KSxcbiAgICApO1xuXG4gICAgbG9nVGV4dC55ID0gODA7XG4gICAgbG9nVGV4dC54ID0gMjtcblxuICAgIGFwcC5zdGFnZS5uYW1lID0gJ3N0YWdlJztcblxuICAgIC8vIE1vdW50IG91dGVyIGJsYWNrIGJveFxuICAgIGNvbnN0IGJsYWNrQm94ID0gYXBwLnN0YWdlLmFkZENoaWxkKG5ldyBHcmFwaGljcygpLnJlY3QoMCwgMCwgNDAwLCA0MDApLmZpbGwoeyBjb2xvcjogMCB9KSk7XG5cbiAgICBibGFja0JveC5uYW1lID0gJ2JsYWNrIGJveCc7XG4gICAgYmxhY2tCb3gueCA9IDQwMDtcblxuICAgIC8vIE1vdW50IHdoaXRlIGJveCBpbnNpZGUgdGhlIHdoaXRlIG9uZVxuICAgIGNvbnN0IHdoaXRlQm94ID0gYmxhY2tCb3guYWRkQ2hpbGQobmV3IEdyYXBoaWNzKCkucmVjdCgxMDAsIDEwMCwgMjAwLCAyMDApLmZpbGwoeyBjb2xvcjogMHhmZmZmZmYgfSkpO1xuXG4gICAgd2hpdGVCb3gubmFtZSA9ICd3aGl0ZSBib3gnO1xuXG4gICAgLy8gRW5hYmxlIGludGVyYWN0aXZpdHkgZXZlcnl3aGVyZSFcbiAgICBhcHAuc3RhZ2UuZXZlbnRNb2RlID0gJ3N0YXRpYyc7XG4gICAgYXBwLnN0YWdlLmhpdEFyZWEgPSBhcHAuc2NyZWVuO1xuICAgIHdoaXRlQm94LmV2ZW50TW9kZSA9ICdkeW5hbWljJztcbiAgICBibGFja0JveC5ldmVudE1vZGUgPSAnc3RhdGljJztcblxuICAgIGZ1bmN0aW9uIG9uRXZlbnQoZSlcbiAgICB7XG4gICAgICAgIGNvbnN0IHR5cGUgPSBlLnR5cGU7XG4gICAgICAgIGNvbnN0IHRhcmdldE5hbWUgPSBlLnRhcmdldC5uYW1lO1xuICAgICAgICBjb25zdCBjdXJyZW50VGFyZ2V0TmFtZSA9IGUuY3VycmVudFRhcmdldC5uYW1lO1xuXG4gICAgICAgIC8vIEFkZCBldmVudCB0byB0b3Agb2YgbG9nc1xuICAgICAgICBsb2dzLnB1c2goYCR7Y3VycmVudFRhcmdldE5hbWV9IHJlY2VpdmVkICR7dHlwZX0gZXZlbnQgKHRhcmdldCBpcyAke3RhcmdldE5hbWV9KWApO1xuXG4gICAgICAgIGlmIChjdXJyZW50VGFyZ2V0TmFtZSA9PT0gJ3N0YWdlJyB8fCB0eXBlID09PSAncG9pbnRlcmVudGVyJyB8fCB0eXBlID09PSAncG9pbnRlcmxlYXZlJylcbiAgICAgICAge1xuICAgICAgICAgICAgbG9ncy5wdXNoKCctLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLScsICcnKTtcbiAgICAgICAgfVxuXG4gICAgICAgIC8vIFByZXZlbnQgbG9ncyBmcm9tIGdyb3dpbmcgdG9vIGxvbmdcbiAgICAgICAgaWYgKGxvZ3MubGVuZ3RoID4gMzApXG4gICAgICAgIHtcbiAgICAgICAgICAgIHdoaWxlIChsb2dzLmxlbmd0aCA%2BIDMwKVxuICAgICAgICAgICAge1xuICAgICAgICAgICAgICAgIGxvZ3Muc2hpZnQoKTtcbiAgICAgICAgICAgIH1cbiAgICAgICAgfVxuXG4gICAgICAgIC8vIFVwZGF0ZSBsb2dUZXh0XG4gICAgICAgIGxvZ1RleHQudGV4dCA9IGxvZ3Muam9pbignXFxuJyk7XG4gICAgfVxuXG4gICAgW2FwcC5zdGFnZSwgd2hpdGVCb3gsIGJsYWNrQm94XS5mb3JFYWNoKChvYmplY3QpID0%2BXG4gICAge1xuICAgICAgICBvYmplY3QuYWRkRXZlbnRMaXN0ZW5lcigncG9pbnRlcmVudGVyJywgb25FdmVudCk7XG4gICAgICAgIG9iamVjdC5hZGRFdmVudExpc3RlbmVyKCdwb2ludGVybGVhdmUnLCBvbkV2ZW50KTtcbiAgICAgICAgb2JqZWN0LmFkZEV2ZW50TGlzdGVuZXIoJ3BvaW50ZXJvdmVyJywgb25FdmVudCk7XG4gICAgICAgIG9iamVjdC5hZGRFdmVudExpc3RlbmVyKCdwb2ludGVyb3V0Jywgb25FdmVudCk7XG4gICAgfSk7XG59KSgpO1xuIn0sIi9zcmMvc3R5bGVzLmNzcyI6eyJjb2RlIjoiYm9keSB7IG1hcmdpbjogMDsgb3ZlcmZsb3c6IGhpZGRlbjsgfSJ9LCIvaW5kZXguaHRtbCI6eyJjb2RlIjoiXG48IURPQ1RZUEUgaHRtbD5cbiAgPGh0bWw%2BXG4gIDxoZWFkPlxuICA8dGl0bGU%2BUGl4aUpTIFBsYXlncm91bmQ8L3RpdGxlPlxuICA8bWV0YSBjaGFyc2V0PVwiVVRGLThcIiAvPlxuICA8bGluayByZWw9XCJzdHlsZXNoZWV0XCIgaHJlZj1cInNyYy9zdHlsZXMuY3NzXCI%2BXG48L2hlYWQ%2BXG48Ym9keT5cbiAgPHNjcmlwdCBzcmM9XCJzcmMvaW5kZXguanNcIj48L3NjcmlwdD5cbjwvYm9keT5cbjwvaHRtbD4ifSwiL3BhY2thZ2UuanNvbiI6eyJjb2RlIjoie1xuICBcInNjcmlwdHNcIjoge1xuICAgIFwic3RhcnRcIjogXCJwYXJjZWwgaW5kZXguaHRtbCAtLW9wZW5cIixcbiAgICBcImJ1aWxkXCI6IFwicGFyY2VsIGJ1aWxkIGluZGV4Lmh0bWxcIlxuICB9LFxuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJwaXhpLmpzXCI6IFwiOC4wLjJcIlxuICB9LFxuICBcImRldkRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJwYXJjZWwtYnVuZGxlclwiOiBcIl4xLjYuMVwiLFxuICAgIFwiQGJhYmVsL2NvcmVcIjogXCJeNy4yMS4zXCIsXG4gICAgXCJAYmFiZWwvcGx1Z2luLXByb3Bvc2FsLWNsYXNzLXByb3BlcnRpZXNcIjogXCJeNy4xMC4xXCJcbiAgfSxcbiAgXCJtYWluXCI6IFwiL2luZGV4Lmh0bWxcIlxufSJ9LCIvLmJhYmVscmMiOnsiY29kZSI6IntcbiAgXCJwcmVzZXRzXCI6IFtcbiAgICBbXG4gICAgICBcIkBiYWJlbC9wcmVzZXQtZW52XCIsXG4gICAgICB7XG4gICAgICAgIFwidXNlQnVpbHRJbnNcIjogXCJlbnRyeVwiLFxuICAgICAgICBcImNvcmVqc1wiOiBcIjMuMjlcIixcbiAgICAgICAgXCJ0YXJnZXRzXCI6IFwibGFzdCAyIGNocm9tZSB2ZXJzaW9uc1wiXG4gICAgICB9XG4gICAgXVxuICBdLFxuICBcInBsdWdpbnNcIjogW1xuICAgIFwiQGJhYmVsL3BsdWdpbi1wcm9wb3NhbC1jbGFzcy1wcm9wZXJ0aWVzXCJcbiAgXVxufSJ9LCIvc2FuZGJveC5jb25maWcuanNvbiI6eyJjb2RlIjoie1xuICAgICAgICAgICAgICAgIFwiaW5maW5pdGVMb29wUHJvdGVjdGlvblwiOiBmYWxzZVxuICAgICAgICAgICAgfSJ9fSwidmlzaWJsZUZpbGVzIjpbIi9zcmMvaW5kZXguanMiXSwiYWN0aXZlRmlsZSI6Ii9zcmMvaW5kZXguanMifQ%3D%3D&exampleId=custom)

Another simple example: https://www.pixiplayground.com/#/edit/exDzbrsXsLvgdbWm0iZFG

### Environment

- **`pixi.js` version**: 8.1.0
- **Browser & Version**: Version 123.0.6312.87 (Official Build) (arm64)
- **OS & Version**: Mac OS Ventura 13.2.1 (22D68)
- **Running Example**: https://www.pixiplayground.com/#/edit/exDzbrsXsLvgdbWm0iZFG

**Comment by CatchABus** [comment]
The root cause seems to be here: https://github.com/pixijs/pixijs/blob/dev/src/events/EventTicker.ts#L110
That dispatch might need better handling.

**Comment by Nantris** [comment]
Great find @CatchABus.

@danielbarion is this not a duplicate of #10131. If it is, I'd encourage you to close this one and join us in discussion over there, but I understand it may be a distinct (albeit related) issue.

**Comment by danielbarion** [comment]
@Nantris I saw on the issue you mentioned the following: `The issue occurs about 90% of the time.`.

Here the issue happens 100% of the time when we have static event mode and dynamic in the same scenario/app.

I had the issue and a friend from Pixi.js team identified the issue, he asked me to create this issue so the team can work on it when they have time.

So I believe it's better to keep this one open and let the team close it, thanks for sharing the other issue
